### PR TITLE
Remove experimental JavaScript plugins

### LINKED ISSUE #15713: Remove deprecated JavaScript plugins
`coffeescript-base`, `envjs`, `javascript-base`, `jshint`, `rhino`.

### Checklist
- [x] Validate whether we should remove the API in the 7.0 release
  - If the removal is not possible then create a follow-up issue and link it in the [overview spreadsheet](https://docs.google.com/spreadsheets/d/19J1nR_dFKpfKdu5KDFMVZGfjR0ysT9DthsBUPwf8mkM/edit#gid=493905624) 
- [x] Update the upgrade guide
- [x] Check User Manual for dangling references and outdated documentation
- [ ] Mark the story as done in the [overview spreadsheet](https://docs.google.com/spreadsheets/d/19J1nR_dFKpfKdu5KDFMVZGfjR0ysT9DthsBUPwf8mkM/edit#gid=2040581133).

### PR DESCRIPTION
**Pull request by jjohannes**
Fixes #15713

### CODE REVIEW

**Comment by jjohannes** [comment]
@bot-gradle test this

**Comment by jjohannes** [comment]
@bot-gradle test this

File: subprojects/docs/src/docs/userguide/migration/upgrading_version_6.adoc [hunk 1]
```diff
@@ -70,6 +70,13 @@ Please refer to the <<publishing_maven.adoc#publishing_maven, documentation of t
 The deprecated Play plugins have been removed.
 An external replacement, the link:https://gradle.github.io/playframework[Play Framework plugin], is available from the plugin portal.
 
+==== Removal of experimental JavaScript plugins
+
+The following bundled plugins for experimental JavaScript integration were never announced and are now removed from the distribution:
```
Comment (melix) [inline subprojects/docs/src/docs/userguide/migration/upgrading_version_6.adoc:75]: :o: You could remove the `never announced` part. It doesn't really help, except for relieving us :)

**Review by melix — APPROVED** [review]
Nothing breaks with the binary compatibility changes check?

**Comment by jjohannes** [comment]
> Nothing breaks with the binary compatibility changes check?

This is because the whole Jar is gone and the list of Jars of the current distribution seems to steer this. Something we could improve in the check. In this case, we know what we are doing. So I think it is ok.

**Comment by jjohannes** [comment]
@bot-gradle test this

**Comment by jjohannes** [comment]
Accidentally, a class for "javascript" is used here: https://github.com/gradle/playframework/blob/72c5001c38cfb8ae2f8bb214f6404e60877ac089/src/main/java/org/gradle/playframework/tools/internal/javascript/GoogleClosureCompiler.java#L12

Will have to fix the plugin to make the smoke tests pass.

**Comment by jjohannes** [comment]
@bot-gradle test this

**Pull request merged**